adapters/repos/db/migrator.go

```diff
@@ -150,24 +150,52 @@ func (m *Migrator) UpdateShardStatus(ctx context.Context, className, shardName,
 	return idx.updateShardStatus(ctx, shardName, targetStatus)
 }
 
-func (m *Migrator) AddPartitions(ctx context.Context, class *models.Class, names []string) error {
+// CreatePartitions creates new partitions and returns a commit func
+// that can be used to either commit or rollback the partitions
+func (m *Migrator) NewPartitions(ctx context.Context, class *models.Class, partitions []string) (commit func(success bool), err error) {
```

Comment:
Probably you want to update this comment with the correct method name

Reply:
Absolutely 👍